exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/limit/TestLimitWithExchanges.java

```diff
@@ -26,4 +30,34 @@
   public void testLimitWithExchanges() throws Exception{
     testPhysicalFromFile("limit/limit_exchanges.json");
   }
+
+  @Test
+  public void testPushLimitPastUnionExchange() throws Exception {
```

Comment:
I added more unit tests with data validation. The data validation is only to verify row count, but not the record. That's because depending on which scan minor fragment first send the data, the final result could be changing.